### PR TITLE
Support for Wavefront Dropwizard Metrics reporter

### PR DESCRIPTION
**Pull request by sushantdewan123**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,120 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.wavefront</groupId>
+    <artifactId>dropwizard-metrics</artifactId>
```
Comment (vikramraman) [inline pom.xml]: There is already a `dropwizard-metrics` library published under `com.wavefront` currently at version 4.30: http://mvnrepository.com/artifact/com.wavefront/dropwizard-metrics

Should be okay continuing that if we bump up version to > 4.30, but we should rename if we want this to be a brand new artifact.

File: src/main/java/com/codahale/metrics/WavefrontHistogram.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+import com.google.common.annotations.VisibleForTesting;
+
+import java.io.OutputStream;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * WavefrontHistogram implementation for com.codahale.metrics
+ * Caveat: Cannot use the same WavefrontHistogram registry for multiple reporters as the reporter
+ * will change the state of the DeltaCounter every time the value is reported.
```
Comment (vikramraman) [inline src/main/java/com/codahale/metrics/WavefrontHistogram.java]: Doc: DeltaCounter or WavefrontHistogram here?

File: src/main/java/com/wavefront/dropwizard/metrics/DropwizardMetricsReporter.java [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+
+import static com.wavefront.sdk.common.Constants.DELTA_PREFIX;
+
+/**
+ * A reporter which publishes metric values to a Wavefront Proxy from a
```
Comment (vikramraman) [inline src/main/java/com/wavefront/dropwizard/metrics/DropwizardMetricsReporter.java]: Tweak doc to convey support for both proxy and direct ingestion.

**Review by vikramraman — CHANGES_REQUESTED** [review]
Just some minor comments. LGTM otherwise.

File: src/main/java/com/codahale/metrics/WavefrontHistogram.java [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    delegate = new com.wavefront.sdk.entities.histograms.WavefrontHistogram(clockMillis);
+  }
+
+  @Override
+  public void update(int value) {
+    delegate.update(value);
+  }
+
+  @Override
+  public void update(long value) {
```
Comment (hanwavefront) [inline src/main/java/com/codahale/metrics/WavefrontHistogram.java:57]: Should we provide an update(double value) method as well?

File: src/main/java/com/codahale/metrics/WavefrontHistogram.java [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+/**
+ * WavefrontHistogram implementation for com.codahale.metrics
+ * Caveat: Cannot use the same WavefrontHistogram registry for multiple reporters as the reporter
+ * will change the state of the DeltaCounter every time the value is reported.
+ *
+ * @author Sushant Dewan (sushant@wavefront.com).
+ */
+public class WavefrontHistogram extends Histogram implements Metric {
+
+  private final com.wavefront.sdk.entities.histograms.WavefrontHistogram delegate;
```
Comment (hanwavefront) [inline src/main/java/com/codahale/metrics/WavefrontHistogram.java]: Nit: import vs full qualified name?

File: pom.xml (hunk 1 above)
Comment (sushantdewan123) [reply pom.xml]: I will call the new one "dropwizard-metrics-sdk"

File: src/main/java/com/codahale/metrics/WavefrontHistogram.java (hunk 5 above)
Comment (sushantdewan123) [reply src/main/java/com/codahale/metrics/WavefrontHistogram.java]: It actually collides with current com.codahale.metrics.WavefrontHistogram vs 
com.wavefront.sdk.entities.histograms.WavefrontHistogram

Maybe we should rename
com.wavefront.sdk.entities.histograms.WavefrontHistogram to
"WavefrontHistogramImpl"

File: src/main/java/com/codahale/metrics/WavefrontHistogram.java (hunk 4 above)
Comment (sushantdewan123) [reply src/main/java/com/codahale/metrics/WavefrontHistogram.java:57]: Yes, can add that but not supported in com.codahale.metrics.Histogram
so won't be overridden

**Review by vikramraman — APPROVED** [review]

**Pull request merged**